_posts/2023-11-01-#28368.md

```diff
@@ -7,18 +7,73 @@ authors: [ismaelsadeeq]
 components: ["tx fees and policy"]
 host: ismaelsadeeq
 status: upcoming
-commit:
+commit: 79bcc5ca0679daf6e57fc4d7ce2244262a7cfd13
 ---
 
-_Notes and questions to follow soon!_
 
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+* The [`CBlockPolicyEstimator`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/policy/fees.h#L146C13-L146C13) is used to estimate fees based on a confirmation target. We have two RPCs, [`estimaterawfees`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/rpc/fees.cpp#L100) and [`estimatesmartfees`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/rpc/fees.cpp#L29) used to obtain a fee estimate for a given confirmation target. Fee estimation is also heavily used in the [wallet](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/wallet/fees.cpp#L29).
+
+* Before this PR [`CTxMempool`](https://github.com/bitcoin/bitcoin/blob/22fa1f4702e9a60e141f545f8d09704deca34b22/src/txmempool.h#L301) owns a [`CBlockPolicyEstimator`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/policy/fees.h#L146C13-L146C13) as one of its members and, whenever a transaction is added or removed from the mempool in [`addUnchecked`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/txmempool.h#L475), [`removedUnchecked`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/txmempool.h#L787) and  [`removeForBlock`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/txmempool.h#L487) methods, it also updates the fee estimator synchronously.
+ 
+* `removeForBlock` calls the fee estimator [`processBlock`](https://github.com/bitcoin/bitcoin/blob/106ab20f121f14d021725c8a657999079dbabfc1/src/policy/fees.h#L204C11-L204C11) to update the fee stats when transactions are removed from the mempool after a new block connection.
+
+* This `removeForBlock` fee estimator updates block [`ConnectTip`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/validation.h#L732C13-L732C13). Adding more steps to `processBlock` can slow down block processing even further.
```

Comment:
Took me a few tries to parse this since "updates" is the noun and "block" is the verb. Maybe some extra words could help:
```suggestion
* Since `removeForBlock` is part of a series of function calls within [`ConnectTip`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/validation.h#L732C13-L732C13), block processing is blocked by fee estimator updates. Adding more steps to `CBlockPolicyEstimator::processBlock` can slow down block processing even further.
```